src/Symfony/Component/HttpKernel/Attribute/MapQueryString.php

```diff
@@ -25,13 +25,17 @@
 class MapQueryString extends ValueResolver
 {
     public ArgumentMetadata $metadata;
+    public readonly string|\Closure|null $validationGroupsResolver;
 
     public function __construct(
         public readonly array $serializationContext = [],
         public readonly string|GroupSequence|array|null $validationGroups = null,
         string $resolver = RequestPayloadValueResolver::class,
         public readonly int $validationFailedStatusCode = Response::HTTP_NOT_FOUND,
+        string|callable $validationGroupsResolver = null,
```

Comment:
What about supporting only service ids?

Reply:
What about an array of validation groups to use instead of a resolver (less boilerplate for simple cases)?
We could also have both options.
Another way could be to have a specific interface to implement on the value object directly.

Reply:
Determining the validation groups dynamically based on the state of the underlying object is already supported by 
`#[Assert\GroupSequenceProvider]` https://symfony.com/doc/current/validation/sequence_provider.html#group-sequence-providers

I guess the use case the author wants to cover here is determining the groups based on external services through DI.